packages/modules/electricity_tariffs/energycharts/tariff.py

```diff
@@ -0,0 +1,37 @@
+from typing import Dict
+from datetime import datetime, timedelta
+from modules.common import req
+
+from modules.common.abstract_device import DeviceDescriptor
+from modules.common.component_state import TariffState
+from modules.common.configurable_tariff import ConfigurableElectricityTariff
+from modules.electricity_tariffs.energycharts.config import EnergyChartsTariffConfiguration
+from modules.electricity_tariffs.energycharts.config import EnergyChartsTariff
+
+
+def fetch_prices(config: EnergyChartsTariffConfiguration) -> Dict[int, float]:
+    current_dateTime = datetime.now()
+    tomorrow = datetime.now() + timedelta(1)
+    start_time = current_dateTime.strftime("%Y-%m-%d") + 'T00%3A00%2B01%3A00'
+    end_time = tomorrow.strftime("%Y-%m-%d") + 'T23%3A59%2B01%3A00'
+    url = f'https://api.energy-charts.info/price?bzn={config.country}&start={start_time}&end={end_time}'
+    add_price = config.serve_price
+    raw_prices = req.get_http_session().get(url).json()
+    time_stamp_arr = []
+    price_arr = []
+    for unix_sec in raw_prices['unix_seconds']:
+        time_stamp_arr.append(unix_sec)  # Epoch from ms in s
+    for price in raw_prices['price']:
+        price_arr.append((float(price + (add_price*10))/1000000))  # €/MWh -> €/Wh + Aufschlag
+    prices: Dict[int, float] = {}
+    prices = dict(zip(time_stamp_arr, price_arr))
```

Comment:
Deine Variable `time_stamp_arr` sieht nach der ersten for-Schleife genau so aus, wie die Quelle `raw_prices["unix_seconds"]`. Die Schleife kann demnach eigentlich wegfallen. Könnte dann recht kompakt so aussehen (ungetestet!):
```suggestion python
    prices = [(timestamp, (price + (config.surcharge * 10)) / 1000000) for timestamp, price in zip(raw_prices["unix_seconds"], raw_prices['price'])]
```

Reply:
Readability

Reply:
Dann bitte zumindest die erste Schleife und die Variable `time_stamp_arr` weglassen. Bei dem Aufruf von `zip()` kannst Du direkt `raw_prices['timestamp']` verwenden.

Reply:
done